### PR TITLE
Write JSON directly to stream in SQL

### PR DESCRIPTION
**Pull request by viliam-durina**

### CODE REVIEW

**Review by gierlachg — APPROVED** [review]

File: hazelcast-jet-sql/src/main/java/com/hazelcast/jet/sql/impl/extract/JsonQueryTarget.java [hunk 1]
```diff
@@ -33,14 +33,13 @@
...
-    private ObjectNode json;
+    private JsonNode json;
 
     @Override
     public void setTarget(Object target) {
         try {
-            json = (ObjectNode) (target instanceof ObjectNode ? target
-                    : target instanceof String ? MAPPER.readTree((String) target)
-                    : MAPPER.readTree((byte[]) target));
+            json = target instanceof ObjectNode ? (ObjectNode) target
```
Comment (gierlachg) [inline hazelcast-jet-sql/src/main/java/com/hazelcast/jet/sql/impl/extract/JsonQueryTarget.java:41]: Tests should be adjusted?

File: hazelcast-jet-sql/src/main/java/com/hazelcast/jet/sql/impl/inject/JsonUpsertTarget.java [hunk 2]
```diff
@@ -16,25 +16,26 @@
...
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
 
 import static com.hazelcast.jet.impl.util.ExceptionUtil.sneakyThrow;
 import static com.hazelcast.jet.sql.impl.inject.UpsertInjector.FAILING_TOP_LEVEL_INJECTOR;
 
 @NotThreadSafe
 class JsonUpsertTarget implements UpsertTarget {
 
-    private static final ObjectMapper MAPPER = new ObjectMapper();
```
Comment (gierlachg) [inline hazelcast-jet-sql/src/main/java/com/hazelcast/jet/sql/impl/inject/JsonUpsertTarget.java:35]: `com.fasterxml.jackson.core:jackson-databind` isn't needed anymore?

File: hazelcast-jet-sql/src/main/java/com/hazelcast/jet/sql/impl/inject/JsonUpsertTarget.java [hunk 3]
```diff
@@ -16,25 +16,26 @@
...
+    private static final JsonFactory JSON_FACTORY = new JsonFactory();
 
-    private ObjectNode json;
+    private final ByteArrayOutputStream baos = new ByteArrayOutputStream();
+    private JsonGenerator jsonGen;
```
Comment (gierlachg) [inline hazelcast-jet-sql/src/main/java/com/hazelcast/jet/sql/impl/inject/JsonUpsertTarget.java:38]: Simply `json`? It's not obvious without checking the type what `gen` is.

File: hazelcast-jet-sql/src/main/java/com/hazelcast/jet/sql/impl/inject/JsonUpsertTarget.java (hunk 2 above)
Comment (viliam-durina) [reply hazelcast-jet-sql/src/main/java/com/hazelcast/jet/sql/impl/inject/JsonUpsertTarget.java:35]: It is, when reading. I only adapted writing.

Comment (viliam-durina) [reply hazelcast-jet-sql/src/main/java/com/hazelcast/jet/sql/impl/inject/JsonUpsertTarget.java:35]: after the last commit it's still needed when writing: we support OBJECT fields of type `JsonNode`, which is from jackson-databind

**Pull request merged**